src/transaction/buffer.rs

```diff
@@ -275,6 +275,31 @@ impl Buffer {
             }
         }
     }
+
+    pub async fn get_write_size(&self) -> usize {
+        let mutations = self.mutations.lock().await;
+        mutations
+            .entry_map
+            .iter()
+            .filter(|(_, v)| {
+                matches!(
+                    v,
+                    BufferEntry::Put(_) | BufferEntry::Insert(_) | BufferEntry::Del
+                )
+            })
+            .map(|(k, v)| {
```

Comment:
You can use `filter_map` instead of `filter(...).map(...)`

Reply:
If we make the heartbeat interval configurable, do you think we should make it as an option in transaction option?

Reply:
Yes, that sounds right to me

Reply:
@ekexium @andylokandy what you think of this?